src/main/src/main/java/org/geoserver/catalog/event/impl/CatalogBeforeAddEventImpl.java

```diff
@@ -0,0 +1,10 @@
+/* (c) 2019 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.catalog.event.impl;
+
+import org.geoserver.catalog.event.CatalogBeforeAddEvent;
+
+/** @author ImranR */
+public class CatalogBeforeAddEventImpl extends CatalogEventImpl implements CatalogBeforeAddEvent {}
```

Comment:
Not sure what's going on here :smile:.

Reply:
So basically we don't need any new methods unlike the Modify events where you need to know the old and new properties. The class basically a wrapper that helps invoking the listener method. This is in line with the implementation of CatalogAddEvent